### PR TITLE
feat(iam): local user password expiration management

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- Added `Get-LocalUserPasswordExpiration` cmdlet to retrieve the password expiration policy for a local user.
- Added `Set-LocalUserPasswordExpiration` cmdlet to configure the password expiration policy for a local user.
- Added `Request-LocalUserPasswordExpiration` cmdlet to retrieve the password expiration policy for a local user from a workload domain.
- Added `Update-LocalUserPasswordExpiration` cmdlet to configure the password expiration policy for a local user from any virtual machine.

Signed-off-by: Gary Blake <gblake@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -16852,8 +16863,185 @@ Function Update-VcenterRootPasswordExpiration {
...
+        .DESCRIPTION
+        The Update-LocalUserPasswordExpiration cmdlet configures a local user password expiration policy. The cmdlet
+        connects to SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to vCenter Server
+		- Configures the local user password expiration policy
+
+        .EXAMPLE
+        Update-LocalUserPasswordExpiration -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -vmName sfo-wsa01 -guestUser root -guestPassword VMw@re1! -localUser "root","sshuser" -minDays 0 -maxDays 999 -warnDays 14
+        This example retrieves the global password expiration policy for the vCenter Server
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:16976]:
```suggestion
        This example updates the global password expiration policy for the vCenter Server
```

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+        .DESCRIPTION
+        The Get-LocalUserPasswordExpiration cmdlets retrieves the password expiration policy for a local user
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-w01-vc01 -guestUser root -guestPassword VMw@re1! -localUser root
+        This example retrieves the password expiration policy for the root user on vCenter Server sfo-w01-vc01
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-w01-nsx01a -guestUser root -guestPassword VMw@re1!VMw@re1! -localUser admin
+        This example retrieves the password expiration policy for the admin user on vCenter Server sfo-w01-nsx01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18086]:
```suggestion
        This example retrieves the password expiration policy for the admin user on NSX Manager sfo-w01-nsx01a
```

File: PowerValidatedSolutions.psm1 (hunk 1 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:16976]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-vcf01 -guestUser root -guestPassword VMw@re1! -localUser vcf
+        This example retrieves the password expiration policy for the vcf user on vCenter Server sfo-vcf01
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-wsa01 -guestUser root -guestPassword VMw@re1! -localUser sshuser
+        This example retrieves the password expiration policy for the vcf user on vCenter Server sfo-vcf01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18094]:
```suggestion
        This example retrieves the password expiration policy for the root user on Workspace ONE Access node sfo-wsa01
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+        Get-LocalUserPasswordExpiration -vmName sfo-w01-vc01 -guestUser root -guestPassword VMw@re1! -localUser root
+        This example retrieves the password expiration policy for the root user on vCenter Server sfo-w01-vc01
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-w01-nsx01a -guestUser root -guestPassword VMw@re1!VMw@re1! -localUser admin
+        This example retrieves the password expiration policy for the admin user on vCenter Server sfo-w01-nsx01
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-vcf01 -guestUser root -guestPassword VMw@re1! -localUser vcf
+        This example retrieves the password expiration policy for the vcf user on vCenter Server sfo-vcf01
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18090]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 3 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18094]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+        .DESCRIPTION
+        The Set-LocalUserPasswordExpiration cmdlets retrieves the password expiration policy for a local user
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-w01-vc01 -guestUser root -guestPassword VMw@re1! -localUser root -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the root user on vCenter Server sfo-w01-vc01
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-w01-nsx01a -guestUser root -guestPassword VMw@re1!VMw@re1! -localUser admin -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the admin user on vCenter Server sfo-w01-nsx01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18137]:
```suggestion
        This example configures the password expiration policy for the root user on NSX Manager sfo-w01-nsx01a
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18137]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 6]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-vcf01 -guestUser root -guestPassword VMw@re1! -localUser vcf -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the vcf user on vCenter Server sfo-vcf01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18141]:
```suggestion
        This example configures the password expiration policy for the vcf user on SDDC Manager sfo-vcf01
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18141]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 7]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-wsa01 -guestUser root -guestPassword VMw@re1! -localUser sshuser -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the vcf user on vCenter Server sfo-vcf01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18145]:
```suggestion
        This example configures the password expiration policy for the sshuser user on Workspace ONE Access sfo-wsa01
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18145]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 4 above)
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18090]:
```suggestion
        This example retrieves the password expiration policy for the vcf user on SDDC Manager sfo-vcf01
```

Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18090]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 8]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+        Get-LocalUserPasswordExpiration -vmName sfo-w01-nsx01a -guestUser root -guestPassword VMw@re1!VMw@re1! -localUser admin
+        This example retrieves the password expiration policy for the admin user on NSX Manager sfo-w01-nsx01a
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-vcf01 -guestUser root -guestPassword VMw@re1! -localUser vcf
+        This example retrieves the password expiration policy for the vcf user on SDDC Manager sfo-vcf01
+
+        .EXAMPLE
+        Get-LocalUserPasswordExpiration -vmName sfo-wsa01 -guestUser root -guestPassword VMw@re1! -localUser sshuser
+        This example retrieves the password expiration policy for the vcf user on Workspace ONE Access sfo-sfo01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18094]:
```suggestion
        This example retrieves the password expiration policy for the sshuser user on Workspace ONE Access sfo-sfo01
```

File: PowerValidatedSolutions.psm1 [hunk 9]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+        .DESCRIPTION
+        The Set-LocalUserPasswordExpiration cmdlets retrieves the password expiration policy for a local user
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-w01-vc01 -guestUser root -guestPassword VMw@re1! -localUser root -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the root user on vCenter Server sfo-w01-vc01
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-w01-nsx01a -guestUser root -guestPassword VMw@re1!VMw@re1! -localUser admin -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the admin user on NSX Manager sfo-w01-nsx01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18137]:
```suggestion
        This example configures the password expiration policy for the admin user on NSX Manager sfo-w01-nsx01a
```

File: PowerValidatedSolutions.psm1 [hunk 10]
```diff
@@ -17881,6 +18069,110 @@ Function Set-VcenterRootPasswordExpiration {
...
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-vcf01 -guestUser root -guestPassword VMw@re1! -localUser vcf -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the vcf user on SDDC Manager sfo-vcf01
+
+        .EXAMPLE
+        Set-LocalUserPasswordExpiration -vmName sfo-wsa01 -guestUser root -guestPassword VMw@re1! -localUser sshuser -minDays 0 -maxDays 999 -warnDays 14
+        This example configures the password expiration policy for the vcf user on Workspace ONE Access sfo-wsa01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:18145]:
```suggestion
        This example configures the password expiration policy for the sshuser user on Workspace ONE Access sfo-wsa01
```

File: PowerValidatedSolutions.psm1 (hunk 8 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18094]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 9 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18137]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 10 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:18145]: Fixed

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**